toolsuite/src/main/java/com/faforever/neroxis/toolsuite/PbrTextureGenerator.java

```diff
@@ -0,0 +1,61 @@
+package com.faforever.neroxis.toolsuite;
+
+import com.faforever.neroxis.cli.OutputFolderMixin;
+import com.faforever.neroxis.cli.VersionProvider;
+import com.faforever.neroxis.util.ImageUtil;
+import picocli.CommandLine;
+
+import javax.imageio.ImageIO;
+import java.awt.image.BufferedImage;
+import java.nio.file.DirectoryStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.concurrent.Callable;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+@CommandLine.Command(name = "generate-pbr", mixinStandardHelpOptions = true, 
+        description = "Generate the pbr texture from individual height and roughness textures", 
+        versionProvider = VersionProvider.class, usageHelpAutoWidth = true)
+public class PbrTextureGenerator implements Callable<Integer> {
+    @CommandLine.Spec
+    private CommandLine.Model.CommandSpec spec;
+    @CommandLine.Mixin
+    private OutputFolderMixin outputFolderMixin;
+
+    @Override
+    public Integer call() throws Exception {
+        generatePbrTexture();
+        return 0;
+    }
+    
+    public void generatePbrTexture() throws Exception {
+        try (DirectoryStream<Path> stream = Files.newDirectoryStream(outputFolderMixin.getOutputPath())) {
+            BufferedImage pbrTexture = new BufferedImage(4096, 4096, BufferedImage.TYPE_INT_ARGB);
+            for (Path path : stream) {
+                System.out.print("processing " + path + "\n");
+                if (Files.isRegularFile(path)) {
+                    Pattern pattern = Pattern.compile("\\d+");
+                    Matcher matcher = pattern.matcher(path.toFile().getName());
+                    if (matcher.find()) {
+                        BufferedImage image = ImageIO.read(path.toFile());
+                        String numberStr = matcher.group();
+                        int layer = Integer.parseInt(numberStr);
+                        if (path.getFileName().toString().startsWith("Roughness")) {
+                            System.out.print("Writing roughness texture\n");
+                            pbrTexture = ImageUtil.setRoughnessMap(pbrTexture, image, layer);
+                        } else if (path.getFileName().toString().startsWith("Height")) {
+                            System.out.print("Writing height texture\n");
+                            pbrTexture = ImageUtil.setHeightMap(pbrTexture, image, layer);
+                        }
+                    }
+                }
+            }
```

Comment:
Rather than searching for the files with pattern matching we should just have the user pass the files via a commandline option

Reply:
Isn't this quite inconvenient when you have to pass up to 16 files? It creates a rather long input string which is a bit clunky to handle
What is the downside of doing it this way?

Reply:
I would view it a bit more favorably than right now which is an opaque naming scheme one would have to follow with no failure mode in case one is not found

Reply:
At least we should have the user specify the folder or name prefix and throw errors if any assumptions we make about which files are present ends up being false.

Otherwise users just get not the result they want and it is hard to figure out why exactly.

Reply:
At the moment the user specifies the folder where to look for the textures. Should I create an informative log message when I skip a file because it doesn't fit the naming scheme? I agree that we should try to avoid that it is too opaque, but at the moment I am not sure what kind of solution you have in mind. Or where we should inform the user